File: RdFinanceBD/src.cpp/MortgageDuration.cpp
#include "MortgageDuration.h"

namespace ca
{
	namespace mss
	{
		namespace rd
		{
			namespace mortgage
			{
				namespace impl
				{
					using ca::mss::rd::mortgage::PaymentFrequency;

					MortgageDuration::MortgageDuration(int years, int month) : years(years), month(month)
					{
					}

					int MortgageDuration::getPaymentsNo(PaymentFrequency *pf)
					{
						return getPaymentsNo(years_Renamed, month, pf);
					}

					int MortgageDuration::monthes()
					{
						return years_Renamed*12 + month;
					}

					int MortgageDuration::years()
					{
						return years_Renamed + month / 12;
					}

					int MortgageDuration::getPaymentsNo(int years, int month, PaymentFrequency *pf)
					{
						return (years + month / 12)*pf->paymentsPerYear + (month % 12)*pf->paymentsPerMonth;
					}

					std::wstring MortgageDuration::toString()
					{
						return (years_Renamed > 0? years_Renamed + std::wstring(L" y"): L"") + (month > 0? month + std::wstring(L" mo"): L"");
					}
				}
			}
		}
	}
}

File: RdFinanceBD/src/ca/mss/rd/mortgage/impl/MortgageDuration.java
package ca.mss.rd.mortgage.impl;

import ca.mss.rd.mortgage.PaymentFrequency;

public class MortgageDuration {
	
	final public int years, month;

	public MortgageDuration(int years, int month) {
		this.years = years;
		this.month = month;
	}
	
	final public int getPaymentsNo(PaymentFrequency pf){
		return getPaymentsNo(years, month, pf);
	}
	
	final public int monthes(){
		return years*12 + month;
	}
	
	final public int years(){
		return years + month/12;
	}
	
	final static public int getPaymentsNo(int years, int month, PaymentFrequency pf){
		return (years+month/12)*pf.paymentsPerYear + (month%12)*pf.paymentsPerMonth.intValue();
	}

	@Override
	public String toString() {
		return (years>0? years+" y": "")+(month>0? month+" mo": "");
	}
	
	
}
